packages/react-component-library/src/components/Field/Field.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useMemo } from 'react'
+
+import { ComponentWithClass } from '../../common/ComponentWithClass'
+import { StyledField } from './partials/StyledField'
+import { StyledError } from './partials/StyledError'
+import { StyledHintText } from './partials/StyledHintText'
+import { getId } from '../../helpers'
+
+export type ErrorType = {
+  error: boolean | string
+}
+
+export interface FieldProps extends ComponentWithClass {
+  hintText?: string
+  errors?: ErrorType[]
+}
+
+export function ariaAttributes(
+  isInvalid: boolean,
+  fieldId: string
+): Record<string, string | boolean> {
+  return {
+    'aria-describedby': fieldId,
+    ...(isInvalid && {
+      'aria-errormessage': `${fieldId}-errormessage`,
+      'aria-invalid': true,
+    }),
+  }
+}
+
+export const Field: React.FC<FieldProps> = ({
+  children,
+  errors,
+  hintText,
+  className,
+}) => {
+  const isInvalid: boolean = useMemo(() => {
+    return errors?.some(({ error }) => error) ?? false
+  }, [errors])
+
+  const fieldId: string = useMemo(() => getId('field'), [])
+
+  return (
+    <StyledField className={className}>
+      {isInvalid &&
+        errors.map(({ error }) => {
```

Comment:
Hmm this will be an error will strict null checks on – might be easier to just to `errors = []` on line 33 rather than `errors?` everywhere.